### PR TITLE
fix(packaging): updated naming scheme for centreon plugins perl dependencies

### PR DESCRIPTION
**Pull request by paul-oureib**
## Description

fixes the packaging of perl dependencies on debian so that bullseye and bookworm generated packages do not have the same name and thus preventing one from replacing the other
currently blocking the debian 12 packaging on web since the dockerize step requires these dependencies and can only find them for one debian distribution

**Fixes** # MON-34667

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

N/A

<h2> How this pull request can be tested ? </h2>

running the perl dependencies workflows and checking if they are delivered with the correct name on artifactory

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/perl-openwsman.yml [hunk 1]
```diff
@@ -164,11 +166,13 @@ jobs:
           distrib: ${{ matrix.distrib }}
           package_extension: ${{ matrix.package_extension }}
           arch: ${{ matrix.arch }}
+          release: 2
```
Comment (kduret) [inline .github/workflows/perl-openwsman.yml:169]: perl-openwsman last release was `3`
```suggestion
          release: 4
```

**Review by kduret — APPROVED** [review]

File: .github/actions/package-nfpm/action.yml [hunk 2]
```diff
@@ -70,23 +83,28 @@ runs:
...
-        MAJOR_LEFT=$( echo $MAJOR_VERSION | cut -d "." -f1 )
-        MAJOR_RIGHT=$( echo $MAJOR_VERSION | cut -d "-" -f1 | cut -d "." -f2 )
-        BUMP_MAJOR_RIGHT=$(( MAJOR_RIGHT_PART + 1 ))
-        if [ "$MAJOR_RIGHT" = "04" ]; then
-          BUMP_MAJOR_LEFT="$MAJOR_LEFT"
-          BUMP_MAJOR_RIGHT="10"
-        else
-          BUMP_MAJOR_LEFT=$(( $MAJOR_LEFT + 1 ))
-          BUMP_MAJOR_RIGHT="04"
+        if [[ "$MAJOR_VERSION" != "" ]]; then
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:95]: -z

File: .github/actions/package-nfpm/action.yml [hunk 3]
```diff
@@ -70,23 +83,28 @@ runs:
...
+          MAJOR_LEFT=$( echo $MAJOR_VERSION | cut -d "." -f1 )
+          MAJOR_RIGHT=$( echo $MAJOR_VERSION | cut -d "-" -f1 | cut -d "." -f2 )
+          if [ "$MAJOR_RIGHT" = "04" ]; then
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:98]:
```suggestion
          if [ "$MAJOR_RIGHT" == "04" ]; then
```

**Review by sc979 — DISMISSED** [review]

File: .github/actions/package-nfpm/action.yml [hunk 4]
```diff
@@ -59,8 +65,15 @@ runs:
...
       run: |
-        export MAJOR_VERSION="${{ inputs.major_version }}"
-        export VERSION="${{ inputs.major_version }}.${{ inputs.minor_version }}"
+        if [ -z ${{ inputs.version }} ]; then
+          export VERSION="${{ inputs.major_version }}.${{ inputs.minor_version }}"
+          export MAJOR_VERSION="${{ inputs.major_version }}"
+          export MINOR_VERSION="${{ inputs.minor_version }}"
+        elif [ -z ${{ inputs.major_version }} ]; then
+          export VERSION="${{ inputs.version }}"
+          export MAJOR_VERSION=$( echo $MAJOR_VERSION | cut -d "-" -f1 )
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:74]:
```suggestion
          export MAJOR_VERSION=$( echo $VERSION | cut -d "-" -f1 )
```

File: .github/actions/package-nfpm/action.yml [hunk 5]
```diff
@@ -59,8 +65,15 @@ runs:
...
+          export MINOR_VERSION=$( echo $MAJOR_VERSION | cut -d "-" -f2 )
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:75]:
```suggestion
          export MINOR_VERSION=$( echo $VERSION | cut -d "-" -f2 )
```

File: .github/actions/package-nfpm/action.yml [hunk 6]
```diff
@@ -70,23 +83,28 @@ runs:
           export APACHE_GROUP="apache"
         else
           export DIST=""
+          if [ "${{ inputs.stability }}" = "unstable" ] || [ "${{ inputs.stability }}" = "canary" ]; then
+            export RELEASE="$RELEASE~${{ inputs.distrib }}"
+          elif [ "${{ inputs.stability }}" = "testing" ]; then
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:88]:
```suggestion
          elif [ "${{ inputs.stability }}" == "testing" ]; then
```

File: .github/actions/package-nfpm/action.yml [hunk 7]
```diff
@@ -70,23 +83,28 @@ runs:
...
-        MAJOR_RIGHT=$( echo $MAJOR_VERSION | cut -d "-" -f1 | cut -d "." -f2 )
-        BUMP_MAJOR_RIGHT=$(( MAJOR_RIGHT_PART + 1 ))
-        if [ "$MAJOR_RIGHT" = "04" ]; then
-          BUMP_MAJOR_LEFT="$MAJOR_LEFT"
-          BUMP_MAJOR_RIGHT="10"
-        else
-          BUMP_MAJOR_LEFT=$(( $MAJOR_LEFT + 1 ))
-          BUMP_MAJOR_RIGHT="04"
+        if [ -z "$MAJOR_VERSION" ]; then
+          MAJOR_LEFT=$( echo $MAJOR_VERSION | cut -d "." -f1 )
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:96]:
```suggestion
          MAJOR_LEFT=$( echo $VERSION | cut -d "." -f1 )
```

File: .github/actions/package-nfpm/action.yml [hunk 8]
```diff
@@ -70,23 +83,28 @@ runs:
...
+          MAJOR_RIGHT=$( echo $MAJOR_VERSION | cut -d "-" -f1 | cut -d "." -f2 )
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:97]:
```suggestion
          MAJOR_RIGHT=$( echo $VERSION | cut -d "-" -f1 | cut -d "." -f2 )
```

File: .github/actions/package-nfpm/action.yml [hunk 9]
```diff
@@ -37,6 +40,9 @@ inputs:
   rpm_gpg_signing_passphrase:
     description: The rpm gpg signing passphrase
     required: true
+  stability:
```
Comment (sc979) [inline .github/actions/package-nfpm/action.yml:43]: at line 24, release seems pretty much mandatory 🤔

**Review by sc979 — APPROVED** [review]

**Review by lucie-dubrunfaut — APPROVED** [review]

**Pull request merged**